lib/hammer_cli_foreman_remote_execution/job_invocation.rb

```diff
@@ -0,0 +1,112 @@
+require 'pry'
+
+module HammerCLIForemanRemoteExecution
+  class JobInvocation < HammerCLIForeman::Command
+    resource :job_invocations
+
+    class ListCommand < HammerCLIForeman::ListCommand
+      output do
+        field :id, _('Id')
+        field :job_name, _('Name')
+        field :state, _('Task State')
+      end
+
+      def extend_data(invocation)
+        JobInvocation.extend_data(invocation)
+      end
+
+      build_options
+    end
+
+    class InfoCommand < HammerCLIForeman::InfoCommand
+      output ListCommand.output_definition do
+        field :hosts, _('Hosts')
+      end
+
+      def extend_data(invocation)
+        JobInvocation.extend_data(invocation)
+      end
+
+      build_options do |o|
+        o.expand(:none)
+      end
+    end
+
+    class OutputCommand < HammerCLIForeman::Command
+      action :output
```

Comment:
----- Original Message -----
> > +    class InfoCommand < HammerCLIForeman::InfoCommand
> > +      output ListCommand.output_definition do
> > +        field :hosts, _('Hosts')
> > +      end
> > +
> > +      def extend_data(invocation)
> > +        JobInvocation.extend_data(invocation)
> > +      end
> > +
> > +      build_options do |o|
> > +        o.expand(:none)
> > +      end
> > +    end
> > +
> > +    class OutputCommand < HammerCLIForeman::Command
> > +      action :output
> 
> The hammer docs theoretically say I can prevent this with `build_options {
> |o| o.expect(:name) }` but it doesn't work.  On IRC, the hammer people (I
> forget who), told me I probably need to write an entire Id resolver
> implementation like hammer-cli-katello, which is very unfortunate.

I would rather track this as a bug and fixing in hammer rather than us duplicating the
code.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/hammer_cli_foreman_remote_execution/pull/1/files#r47103498